### PR TITLE
Add message publishing code to trigger

### PR DESCRIPTION
**Pull request by tapishr**

### CODE REVIEW

File: components/AppBar.tsx [hunk 1]
```diff
@@ -3,17 +3,15 @@ import Banner from "./Banner";
...
 	backdropClassName?: string,
 }>) => {
 	const headerRef = useRef<HTMLDivElement>(null);
-	const appbarRef = useRef<HTMLDivElement>(null);
 	const offsetRef = useRef<HTMLDivElement>(null);
 
 	useEffect(() => {
 		if (headerRef.current && offsetRef.current) {
-			const offsetHeight = headerRef.current.clientHeight - ((!props.offset && appbarRef.current) ? appbarRef.current.clientHeight : 0);
+			const offsetHeight = headerRef.current.clientHeight;
```
Comment (MuskanPaliwal) [inline components/AppBar.tsx:14]: is this change relevant for this PR?

**Review by MuskanPaliwal — APPROVED** [review]
You might want to take a pull from main I guess.

File: pages/api/extension/trigger.ts [hunk 2]
```diff
@@ -20,11 +21,11 @@ export default async function triggeHandler(req: NextApiRequest, res: NextApiRes
 		console.error("[extension/triggeHandler] Error getting user token", err);
 		return null;
 	});
-    if (!user?.email) {
+    if (!user || !user.email) {
```
Comment (avikalpg) [inline pages/api/extension/trigger.ts:24]: Did the optional chaining not handle some situation?

**Review by avikalpg — APPROVED** [review]

**Review by MuskanPaliwal — APPROVED** [review]

File: components/AppBar.tsx (hunk 1 above)
Comment (tapishr) [reply components/AppBar.tsx]: it wasn't. This happened as main was not merged into this branch, not an issue now

File: pages/api/extension/trigger.ts (hunk 2 above)
Comment (tapishr) [reply pages/api/extension/trigger.ts]: This was a bad diff, hadn't merged main into this branch

File: pages/api/extension/trigger.ts [hunk 3]
```diff
@@ -5,29 +5,29 @@ import { DbUser, getUserByAlias } from '../../../utils/db/users';
...
     }
     const user = await getToken({ req: req }).catch((err) => {
-		console.error("[extension/triggeHandler] Error getting user token", err);
-		return null;
-	});
-    if (!user || !user.email) {
+        console.error("[extension/triggeHandler] Error getting user token", err);
+        return null;
+    });
+    if (!user?.email) {
```
Comment (avikalpg) [inline pages/api/extension/trigger.ts:24]: Arey? Reverted to this?

**Review by avikalpg — APPROVED** [review]

**Pull request merged**